sim/pokemon.ts

```diff
@@ -1968,7 +1968,7 @@ export class Pokemon {
 		const item = (this.ignoringItem() ? '' : this.item);
 		if (item === 'ironball') return true;
 		// If a Fire/Flying type uses Burn Up and Roost, it becomes ???/Flying-type, but it's still grounded.
```

Comment:
I couldn't find a unit test for this interaction but I don't trust myself to write one, it's beyond my comprehension.

Reply:
Actually, has Ground immunity for Tera Flying Roost actually been tested?

Reply:
https://www.smogon.com/forums/threads/scarlet-violet-battle-mechanics-research.3709545/post-9410489 It has now